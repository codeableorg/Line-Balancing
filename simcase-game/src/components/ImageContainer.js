/** @jsx jsx */
import { jsx } from "@emotion/core";

import tasksJson from "../data/tasks.json";

const frame = {
  position: "relative",
  width: "200px",
  height: "200px"
};

const img = {
  height: "100%",
  width: "100%",
  objectFit: "cover",
  position: "absolute",
  borderRadius: "50%"
};

function ImageContainer({ id }) {
  const image = tasksJson["scenarios"][id]["image"];
  return (
    <div css={frame}>
      <img css={img} src={image.src} alt={image.name} />
    </div>
  );
}

export default ImageContainer;
