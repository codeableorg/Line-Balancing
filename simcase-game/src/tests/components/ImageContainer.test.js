import React from "react";
import { render } from "@testing-library/react";

import ImageContainer from "../../components/ImageContainer";

test("Test change image is correctly loaded", () => {
  const { asFragment } = render(<ImageContainer />);
  expect(asFragment()).toMatchSnapshot();
});
