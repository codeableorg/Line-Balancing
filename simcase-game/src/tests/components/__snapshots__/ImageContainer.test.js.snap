// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test change image is correctly loaded 1`] = `
<DocumentFragment>
  <img
    alt="line-cook"
    class="css-1k2n4ns-Statement"
    src="/assets/img/line-cook.png"
  />
</DocumentFragment>
`;
